### PR TITLE
chore: Fix style issues

### PR DESCRIPTION
**Pull request by erezrokah**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

Related https://github.com/cloudquery/.github/pull/107

Ensures this repo conforms with our new set of rules in https://github.com/cloudquery/.github/pull/107

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run --new-from-rev main` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ linters:
     - errcheck
     - gocritic
     - gofmt
-    - goimports
+    - gci
```
Comment (erezrokah) [inline .golangci.yml:35]: This is incorporated into the `.github` sync workflow in https://github.com/cloudquery/.github/pull/107

File: client/client.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import (
 	"errors"
 
 	"github.com/Azure/azure-sdk-for-go/services/subscription/mgmt/2020-09-01/subscription"
+	// Import all autorest modules
```
Comment (erezrokah) [inline client/client.go:8]: Blank imports require a comments explaining them.
Depending if we do include a meaningful comment, we can keep the rule or remove it.

File: client/client.go [hunk 3]
```diff
@@ -34,8 +35,8 @@ func (c Client) Logger() hclog.Logger {
 }
 
 // SetSubscriptionServices adds a set of services to an azure subscription
-func (c *Client) SetSubscriptionServices(subscriptionId string, services services.Services) {
-	c.services[subscriptionId] = &services
+func (c *Client) SetSubscriptionServices(subscriptionId string, servicesSet services.Services) {
+	c.services[subscriptionId] = &servicesSet
```
Comment (erezrokah) [inline client/client.go:39]: Shadowing

File: client/config.go [hunk 4]
```diff
@@ -6,7 +6,7 @@ type Config struct {
 	Subscriptions []string `hcl:"subscriptions,optional"`
 }
 
-func (c Config) Example() string {
+func (Config) Example() string {
```
Comment (erezrokah) [inline client/config.go:9]: Unused receiver

File: client/errors.go [hunk 5]
```diff
@@ -19,6 +19,10 @@ var errorCodeDescriptions = map[interface{}]string{
 	http.StatusForbidden:  "You are not authorized to perform this operation.",
 }
 
+var (
```
Comment (erezrokah) [inline client/errors.go:22]: Declaration order

File: resources/services/security/settings.go [hunk 6]
```diff
@@ -73,14 +73,13 @@ func fetchSecuritySettings(ctx context.Context, meta schema.ClientMeta, parent *
 				res <- v
 			} else if v, ok := item.AsAlertSyncSettings(); ok {
 				res <- v
-			} else {
-				return fmt.Errorf("unexpected BasicSetting: %#v", item)
 			}
+
+			return fmt.Errorf("unexpected BasicSetting: %#v", item)
```
Comment (erezrokah) [inline resources/services/security/settings.go:78]: Redundant indentation

File: resources/services/monitor/diagnostic_settings.go [hunk 7]
```diff
@@ -249,11 +249,9 @@ func fetchMonitorDiagnosticSettings(ctx context.Context, meta schema.ClientMeta,
 		})
 	}
 
-	if err := g.Wait(); err != nil {
-		return err
-	}
+	err = g.Wait()
```
Comment (erezrokah) [inline resources/services/monitor/diagnostic_settings.go:252]: Redundant if

File: resources/services/security/settings.go [hunk 8]
```diff
@@ -71,7 +71,7 @@ func fetchSecuritySettings(ctx context.Context, meta schema.ClientMeta, parent *
 				res <- v
 			} else if v, ok := item.AsDataExportSettings(); ok {
 				res <- v
-			} else if v, ok := item.AsAlertSyncSettings(); ok {
+			} else if v, ok := item.AsAlertSyncSettings(); ok { // nolint:revive (in this case revive has a false positive on early-return)
```
Comment (erezrokah) [inline resources/services/security/settings.go:74]: `revive` wants to do `!ok return` inside this `else if` which is not that nice either

**Comment by yevgenypats** [comment]
What the advantages of the new linter?

**Comment by erezrokah** [comment]
> What the advantages of the new linter?

Great question 👍 
We enforce these rules manually when reviewing PRs, so I codified them. The rules are based on comments I got from reviewers on a bunch of my PRs.

**Comment by erezrokah** [comment]
> What the advantages of the new linter?

BTW it's the same tool `golanci-lint` with more rules configured

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**